massa-execution-worker/src/execution.rs

```diff
@@ -1451,8 +1460,24 @@ impl ExecutionState {
                     .unwrap_or_default()
                     .0;
 
-                // set the execution context
-                *context_guard!(self) = execution_context;
+                {
+                    let mut context = context_guard!(self);
+                    *context = execution_context;
+
+                    let call_stack_addr = context.get_call_stack();
```

Comment:
I'm not sure how much we want to mimic the classic call behaviour but I would imagine since we are reproducing the fee and coins now that we would also want to push a new element in the call stack as this is something that a builder might want to test in a read only call.

Also as we go the read only and normal versions are starting to have more and more similarities and I believe we should eventually unify those along with what was specified here: https://github.com/massalabs/massa/issues/4484

Reply:
I agree it would be a nice followup, but less urgent than the remaining security stuff